polsieira/Build-Your-Own-Backend
# Create database

#### What's this PR do?

This PR includes the database and a boilerplate for the server. The database is comprised of the initial migration, a seeding file, a csv file, a cleaning file, and the resulting json file. 

* db/migrations/20191119174500_initial.js: creates the schema for two tables, the talks table and the speakers table, and drops tables

* db/seeds/dev/talks.js: pulls data from data/talkData.json and populates tables with default data

* server.js: please ignore, boilerplate for server, will add functionality in create_server branch

* data/cleaner.js: parse csv file, organizes data, and writes json data to file data/talkData.json

#### Where should the reviewer start?

Start by getting an idea of how the data is formatted in either the db/migrations/20191119174500_initial.js or data/cleaner.js.

#### How should this be manually tested?

Create a local database named tedtalks using psql. Install npm dependencies. Migrate database using `knex migrate:latest` and seed data using `knex seed:run`. Then use Postico or the terminal to visually verify data has successfully been seeded

#### Questions:

1. Are there any more data the table should keep track of? Should I add any migrations?
